[BUG] Link from external hash to external hash doesn't compile

#### Steps to reproduce

```sh
$ hc-scaffold link-type
✔ Link from which entry type? · ExternalHash
✔ What name should be given to the link for this hash? · some
✔ Link to which entry type? · Agent
✔ Which role does this agent play in the relationship ? (eg. "creator", "invitee") · other
✔ Should the link be bidirectional? · yes
✔ Can the link be deleted? · yes

Link type scaffolded!

$ npm t

> test
> npm run build:happ && cargo test


> build:happ
> npm run build:zomes && hc app pack workdir --recursive


> build:zomes
> RUSTFLAGS='--cfg getrandom_backend="custom"' cargo build --release --target wasm32-unknown-unknown
...
error[E0308]: mismatched types
  --> dnas/dena/zomes/coordinator/zoma/src/some_to_others.rs:98:47
   |
98 |         if link.target.clone().into_hash() == input.base_some.clone().into_hash() {
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `HoloHash<AnyLinkable>`, found `HoloHash<External>`
   |
   = note: expected struct `HoloHash<AnyLinkable>`
              found struct `HoloHash<hdk::prelude::hash_type::External>`
help: call `Into::into` on this expression to convert `HoloHash<hdk::prelude::hash_type::External>` into `HoloHash<AnyLinkable>`
   |
98 |         if link.target.clone().into_hash() == input.base_some.clone().into_hash().into() {
   |                                                                                   +++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `zoma` (lib) due to 1 previous error
```

**Tasks:**
- Fix bug and simplify comparison:
```rust
if link.target == input.base_some.clone().into() {
```

--- a/src/scaffold/link_type/coordinator.rs
+++ b/src/scaffold/link_type/coordinator.rs
@@ -521,12 +521,8 @@ fn remove_link_handlers(
     let pascal_link_type_name =
         format_ident!("{}", link_type_name(from_referenceable, to_referenceable));
 
-    let from_link_hash_type_code = hash_type_code_from_referenceable(to_referenceable);
-
     let bidirectional_remove = if bidirectional {
         {
-            let from_inverse_hash_type = hash_type_code_from_referenceable(from_referenceable);
-
             quote! {
                 let links = get_links(
                     LinkQuery::try_new(
@@ -535,7 +531,7 @@ fn remove_link_handlers(
                     GetStrategy::default(),
                 )?;
                 for link in links {
-                    if #from_inverse_hash_type == input.#base_field_name.clone().into_hash() {
+                    if link.target == input.#base_field_name.clone().into() {
                         delete_link(link.create_link_hash, GetOptions::default())?;
                     }
                 }
@@ -559,7 +555,7 @@ fn remove_link_handlers(
                 GetStrategy::default(),
             )?;
             for link in links {
-                if #from_link_hash_type_code == input.#target_field_name.clone().into_hash() {
+                if link.target == input.#target_field_name.clone().into() {
                     delete_link(link.create_link_hash, GetOptions::default())?;
                 }
             }
@@ -569,34 +565,23 @@ fn remove_link_handlers(
     }
 }
 
-fn hash_type_code_from_referenceable(referenceable: &Referenceable) -> TokenStream {
-    match referenceable {
-        Referenceable::Agent { .. } => quote! {
-            AgentPubKey::from(
-                link.target.clone()
-                    .into_entry_hash()
-                    .ok_or(wasm_error!(
-                        WasmErrorInner::Guest("No entry_hash associated with link".to_string())
-                    ))?
-            )
-        },
-        Referenceable::ExternalHash { .. } => quote! {
-            link.target.clone().into_hash()
-        },
-        Referenceable::EntryType(_) => {
-            let field_type = referenceable.field_type().to_string();
-            let into_hash_method_name = format_ident!("into_{}", field_type.to_case(Case::Snake));
-            let error_message = format!(
-                "No {} associated with link",
-                field_type.to_case(Case::Lower)
-            );
+#[cfg(test)]
+mod tests {
+    use super::*;
 
-            quote! {
-                link.target
-                    .clone()
-                    .#into_hash_method_name()
-                    .ok_or(wasm_error!(WasmErrorInner::Guest(#error_message.to_string())))?
-            }
-        }
+    #[test]
+    fn external_hash_deletion_compares_any_linkable_hashes() {
+        let from_referenceable = Referenceable::ExternalHash {
+            name: "from".to_string(),
+        };
+        let to_referenceable = Referenceable::ExternalHash {
+            name: "to".to_string(),
+        };
+
+        let handlers = remove_link_handlers(&from_referenceable, &to_referenceable, true);
+        let generated = unparse_pretty(&syn::parse_quote! { #handlers });
+
+        assert!(generated.contains("if link.target == input.target_to.clone().into()"));
+        assert!(generated.contains("if link.target == input.base_from.clone().into()"));
     }
 }

